flink-runtime-web/src/main/java/org/apache/flink/runtime/webmonitor/handlers/JarRunHandler.java

```diff
@@ -125,7 +125,8 @@ public CompletableFuture<JarRunResponseBody> handleRequest(
                         });
     }
 
-    private SavepointRestoreSettings getSavepointRestoreSettings(
+    @VisibleForTesting
+    SavepointRestoreSettings getSavepointRestoreSettings(
```

Comment:
AFAICT you could make testing way easier by making this static and adding a `Logger` and `Configuration` parameter.